packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -539,11 +526,12 @@ - (void)initialize:(FlutterError* __autoreleasing*)error {
   // Allow audio playback when the Ring/Silent switch is set to silent
   [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback error:nil];
 
-  for (NSNumber* textureId in _players) {
-    [_registry unregisterTexture:[textureId unsignedIntegerValue]];
-    [_players[textureId] dispose];
-  }
-  [_players removeAllObjects];
+  [self.playersByTextureId
+      enumerateKeysAndObjectsUsingBlock:^(NSNumber* textureId, FLTVideoPlayer* player, BOOL* stop) {
+        [self.registry unregisterTexture:textureId.unsignedIntegerValue];
```

Comment:
Use `enumerateKeysAndObjectsUsingBlock` instead of looking up the value inside the key loop.